src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionInvokerProvider.cs

```diff
@@ -97,15 +97,24 @@ public void OnProvidersExecuting(ActionInvokerProviderContext context)
             }
 
             var cache = CurrentCache;
-            PageActionInvokerCacheEntry cacheEntry;
 
             IFilterMetadata[] filters;
-            if (!cache.Entries.TryGetValue(actionDescriptor, out cacheEntry))
+            if (!cache.Entries.TryGetValue(actionDescriptor, out PageActionInvokerCacheEntry cacheEntry))
```

Comment:
The action descriptor(s) would still be valid if the compilation expires, but it's just the cached types that need to be invalidated. Thoughts \ concerns?